### PR TITLE
Create a docker build and push workflow for ghcr (Github Container Registry)

### PR DESCRIPTION
**Pull request by seveibar**
### Description

This will automatically push the latest version of freerouting to ghcr.io with the latest tag

Here's an example of what it looks like when pushed: https://github.com/tscircuit/freerouting/pkgs/container/freerouting

It would be good to have version numbers on each docker package eventually as well, but maybe this is a good first/incremental step to just keep pushing up latest?

(If this is annoying to maintain, feel free to close!)

### Usage

You have to authenticate with ghcr.io to use these types of packages. [Here's the guide](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-with-a-personal-access-token-classic)

Basically you login to ghcr.io then you can do this to get the image:

```bash
docker pull ghcr.io/tscircuit/freerouting:master
```

### CODE REVIEW

**Comment by andrasfuchs** [comment]
Thanks @seveibar! That looks like a good start!

I think we should have `:latest`, `:nightly` and the `:x.y.z` version tags defined as well to be future-proof, so I will get back to this later. 😉

**Review by andrasfuchs — CHANGES_REQUESTED** [review]
Please add the tags part to the metadata-action:

tags: |
  type=raw,value=nightly
  type=sha
  
  I can't edit the file for some reason.
